CenterForOpenScience/osf.io
# [Feature] Fix Fangorn GitHub off branch issues  [OSF-4973]

## Purpose

Most Fangorn actions on non-default github branches fail. This fixes a load of them.


## Changes
1. Renaming in non default branch works (add proper branch references to WB calls) (OSF-4973)
2. Downloading files on the off branch now works (OSF-4972)
3. Downloading zip files of off branch (will soon; pending WB fix) work (OSF-7150)
## Side effects
1. Folder creation in off branch now works
2. Copying files from the off branch to another provider now works
## Known issues
1. Moving or copying multiple files at once does not work (dragging a folder should work thought)
   https://openscience.atlassian.net/browse/OSF-5724
## Ticket



https://openscience.atlassian.net/browse/OSF-4973
https://openscience.atlassian.net/browse/OSF-4972
https://openscience.atlassian.net/browse/OSF-7150
[#OSF-4973]
[#OSF-4972]
[#OSF-7150]
## QA Considerations
1. Test file renaming in Github master and off branch
2. Test folder creation in master and off branch
3. Test downloading a file in master and off branch
4. Test downloading a zip of the branch in master and off branch
5. Test combinations of these on the off branch:
   a. Rename a file then move to another provider (this copies, but doesn't move, fyi)
   b. Drag a file into a folder then rename it
   c. Drag a file into a folder then drag it to another provider 
   d. Rename a file and then drag it into a folder

In each of these cases test that the action occurs without error and then refresh the page and ensure that the changes persist. Check both the master and off branch for changes. The typical issue is that folders or files from the off branch will move to the master branch.
